qa/rpc-tests/test.py

```diff
@@ -0,0 +1,235 @@
+        except:
+            fail_list.append(file)
+        
+        elapsed_time = time.time() - start_time
+        logging.info("Elapsed time: %s", time.strftime("%H:%M:%S", time.gmtime(elapsed_time)))
+        stats.append(time.strftime("%H:%M:%S", time.gmtime(elapsed_time)))
+       
+        if p1.returncode != 0:
+            fail_list.append(stats)
+        else:
+            pass_list.append(stats)
+        
+        num_tests_ran += 1
+
+        if args.failfast and len(fail_list) > 0 :
+            break
+    total_elapsed_time = time.time() - total_stime
+
+    logging.info("=====")
+    logging.info("Total tests: %i ", total_tests)
+    logging.info("Tests ran: %i ", num_tests_ran)
+    logging.info("Duration: ~%s ", time.strftime("%H:%M:%S", time.gmtime(total_elapsed_time)))
+    logging.info("Num Fail: %i", len(fail_list) )
+    logging.info("Num Pass: %i", len(pass_list) )
+    logging.info("=====")
+    #add sort routine
+    #maybe cache meta stuff
+    #sanity check failed tests (e.g. clean cache, kill zombie zcashd pids, dump mem info, retry 3 times)
+    logging.info("+ TEST(S) PASSED :")
+    logging.info("------------------")
+    print_test_results(pass_list)
+    logging.info("- TEST(S) FAILED :")
+    logging.info("------------------")
+    print_test_results(fail_list)
+
+def print_test_results(test_list):
+    i =0
+    for test in test_list:
+        logging.info("%s : %s", test[i], test[i+1])
+        i+=2
+
+def opt_list():
+    opt_listbase()
+    opt_listextended()
+
+def opt_listbase():
+    for test_file in BASE_SCRIPTS:
+            print test_file[:-3]
+
+def opt_listextended():
+    for ext_test_file in EXTENDED_SCRIPTS:
+            print ext_test_file[:-3]
+
+def opt_individual(args):
+    print 'files: {}'.format(args.args)
+    run_test_scripts(args.args, args)
+
+def main():
+    parser = argparse.ArgumentParser(description="Zcashd RPC Tester")
+    parser.add_argument('--list', '-l', action='store_true', help='Prints a list of the base and extended RPC tests.')
+    parser.add_argument('--listbase', '-lb', action='store_true', help='Prints a list of the base RPC tests.')
+    parser.add_argument('--listextended', '-le', action='store_true', help='Prints a list of the extended RPC tests.')
+    parser.add_argument('--extended', '-e', action='store_true', help='Run the extended RPC tests.')
+    parser.add_argument('--individual', '-i', action='store_true', help='Run individual RPC tests.')
+    parser.add_argument('--failfast', '-f', action='store_true', help='Stop execution of tests after first failure.')
+    parser.add_argument('--version', '-v', action='version', version='1.0.0' )
+    parser.add_argument('args', nargs=argparse.REMAINDER)
+    args = parser.parse_args()
+
+    if args.list:
+        opt_list()
+    if args.listbase:
+        opt_listbase()
+    if args.listextended:
+        opt_listextended()
+    if args.individual:
+        opt_individual(args)
```

Comment:
Remove the `-i` argument, and instead check if `args.args` is not empty, to decide whether to run provided tests or all tests.